packages/x-components/src/components/suggestions/base-suggestion.vue

```diff
@@ -79,6 +86,20 @@
       };
     }
 
+    /**
+     * Emits the events when the button is clicked.
+     *
+     * @public
+     */
+    protected emitEvents(): void {
+      forEach<Partial<XEventsTypes>>(this.events, (event, payload) => {
```

Comment:
You don't need to specify the generic, it should be inferred from `this.events`
```suggestion
      forEach(this.events, (event, payload) => {
```